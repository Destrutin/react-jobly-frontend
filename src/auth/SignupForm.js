import React, {useState} from "react";
import {useNavigate} from "react-router-dom";

function SignupForm({signup}) {
    const navigate = useNavigate();
    const [formData, setFormData] = useState({
        username: "",
        password: "",
        firstName: "",
        lastName: "",
        email: "",
    });

    const handleChange = e => {
        const {name, value} = e.target;
        setFormData(data => ({...data, [name]:value}))
    }

    const handleSubmit = async(e) => {
        e.preventDefault();

        try{
            await signup(formData);
            navigate("/companies")
        } catch (err) {
            console.error(err);
        }
    }

    return (
        <div className="SignupForm">
            <h1>Sign Up</h1>
            <div className="card">
                <form onSubmit={handleSubmit}>
                    <div className="form-inputs">
                        <label htmlFor="username">Username</label>
                        <input
                            id="username"
                            name="username"
                            className="form-input"
                            value={formData.username}
                            onChange={handleChange}
                        />
                    </div>
                    <div className="form-inputs">
                        <label htmlFor="password">Password</label>
                        <input
                            id="password"
                            name="password"
                            className="form-input"
                            value={formData.password}
                            onChange={handleChange}
                        />
                    </div>
                    <div className="form-inputs">
                        <label htmlFor="firstName">First name</label>
                        <input
                            id="firstName"
                            name="firstName"
                            className="form-input"
                            value={formData.firstName}
                            onChange={handleChange}
                        />
                    </div>
                    <div className="form-inputs">
                        <label htmlFor="lastName">Last name</label>
                        <input
                            id="lastName"
                            name="lastName"
                            className="form-input"
                            value={formData.lastName}
                            onChange={handleChange}
                        />
                    </div>
                    <div className="form-inputs">
                        <label htmlFor="email">Email</label>
                        <input
                            id="email"
                            name="email"
                            className="form-input"
                            value={formData.email}
                            onChange={handleChange}
                        />
                    </div>
                    <button type="submit" className="submit-button" onSubmit={handleSubmit}>Submit</button>
                </form>
            </div>
        </div>
    )
};

export default SignupForm;