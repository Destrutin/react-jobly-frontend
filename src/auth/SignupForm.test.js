import React from 'react';
import {render, fireEvent} from "@testing-library/react";
import {MemoryRouter} from 'react-router-dom';
import SignupForm from './SignupForm';
import { act } from 'react-dom/test-utils';

test('renders signup form without crashing', () => {
    render(
        <MemoryRouter>
            <SignupForm/>
        </MemoryRouter>
    )
})

test('matches snapshot', () => {
    const testSignup = [{username: "Fang", password: "Fangiscool", firstName: "Fang", lastName: "Boi", email: "FangBoi@gmail.com"}]
    const {asFragment} = render(
        <MemoryRouter>
            <SignupForm signup={testSignup}/>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})

test('Signup form submits with correct data', async () => {
    const mockSignup = jest.fn();

    const {getByLabelText, getByText} = render(
        <MemoryRouter>
            <SignupForm signup={mockSignup}></SignupForm>
        </MemoryRouter>
    )
    
    await act(async () => {
        fireEvent.change(getByLabelText('Username'), {target: {value: 'Fang'}});
        fireEvent.change(getByLabelText('Password'), {target: {value: 'Fangiscool'}});
        fireEvent.change(getByLabelText('First name'), {target: {value: 'Fang'}});
        fireEvent.change(getByLabelText('Last name'), {target: {value: 'Boi'}});
        fireEvent.change(getByLabelText('Email'), {target: {value: 'FangBoi@gmail.com'}});

        fireEvent.submit(getByText('Submit'));

        expect(mockSignup).toHaveBeenCalled();
    })
})