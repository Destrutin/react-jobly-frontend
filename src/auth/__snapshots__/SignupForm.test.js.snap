// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <h1>
      Sign Up
    </h1>
    <div
      class="card"
    >
      <form>
        <div
          class="form-inputs"
        >
          <label
            for="username"
          >
            Username
          </label>
          <input
            class="form-input"
            id="username"
            name="username"
            value=""
          />
        </div>
        <div
          class="form-inputs"
        >
          <label
            for="password"
          >
            Password
          </label>
          <input
            class="form-input"
            id="password"
            name="password"
            value=""
          />
        </div>
        <div
          class="form-inputs"
        >
          <label
            for="firstName"
          >
            First name
          </label>
          <input
            class="form-input"
            id="firstName"
            name="firstName"
            value=""
          />
        </div>
        <div
          class="form-inputs"
        >
          <label
            for="lastName"
          >
            Last name
          </label>
          <input
            class="form-input"
            id="lastName"
            name="lastName"
            value=""
          />
        </div>
        <div
          class="form-inputs"
        >
          <label
            for="email"
          >
            Email
          </label>
          <input
            class="form-input"
            id="email"
            name="email"
            value=""
          />
        </div>
        <button
          class="submit-button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
